library/core/src/option.rs

```diff
@@ -119,7 +119,7 @@
 //! # Representation
 //!
 //! Rust guarantees to optimize the following types `T` such that
-//! [`Option<T>`] has the same size as `T`:
+//! [`Option<T>`] has the same size and alignment as `T`:
```

Comment:
These two words are the core of the change; the other places refer back to here.

Reply:
We actually also need them to be function-call-ABI compatible, right? Even types with the same size and alignment can have different ABIs.

Reply:
That's already guaranteed for a few types, like <https://doc.rust-lang.org/stable/std/num/struct.NonZeroU32.html#layout-1>.

That said, I'd like to keep this PR+FCP to just the hopefully-obvious and non-controversial addition of alignment, and something else can add more.

Reply:
And avoiding deciding on the terminology for this is part of why I'm happy keeping this to just something we have a word for already 🙃